cloud/google/ssh.go

```diff
@@ -19,91 +19,54 @@ package google
 import (
 	"fmt"
 	"io/ioutil"
-	"os"
 	"os/exec"
 	"strings"
 
 	"github.com/golang/glog"
 	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
-const (
-	MachineControllerSshKeySecret = "machine-controller-sshkeys"
-	// Arbitrary name used for SSH.
-	SshUser                = "clusterapi"
-	SshKeyFile             = "clusterapi-key"
-	SshKeyFilePublic       = SshKeyFile + ".pub"
-	SshKeyFilePublicGcloud = SshKeyFilePublic + ".gcloud"
-)
-
-func createSshKeyPairs() error {
-	err := run("ssh-keygen", "-t", "rsa", "-f", SshKeyFile, "-C", SshUser, "-N", "")
-	if err != nil {
-		return fmt.Errorf("couldn't generate RSA keys: %v", err)
-	}
-
-	// Prepare a gce format public key file
-	outfile, err := os.Create(SshKeyFilePublicGcloud)
-	if err != nil {
-		return err
-	}
-	defer outfile.Close()
-
-	b, err := ioutil.ReadFile(SshKeyFilePublic)
-	if err == nil {
-		outfile.WriteString(SshUser + ":" + string(b))
-	}
-
-	return err
-}
-
-func cleanupSshKeyPairs() {
-	os.Remove(SshKeyFile)
-	os.Remove(SshKeyFilePublic)
-	os.Remove(SshKeyFilePublicGcloud)
-}
-
 // It creates secret to store private key.
 func (gce *GCEClient) setupSSHAccess(m *clusterv1.Machine) error {
-	// Create public/private key pairs
-	err := createSshKeyPairs()
+	config, err := gce.providerconfig(m.Spec.ProviderConfig)
 	if err != nil {
 		return err
 	}
 
-	config, err := gce.providerconfig(m.Spec.ProviderConfig)
+	sshKey, err := ioutil.ReadFile(gce.sshCreds.privateKeyPath)
 	if err != nil {
-		return err
+		return fmt.Errorf("Couldn't read ssh cred private key path '%v': %v", gce.sshCreds.privateKeyPath,  err)
```

Comment:
From https://github.com/golang/go/wiki/CodeReviewComments#error-strings, error strings should start with a lowercase character so that multiple errors can be chained together.